dozer-sql/src/pipeline/aggregation/max.rs

```diff
@@ -1,552 +1,78 @@
-//             maximum = dozer_types::rust_decimal::Decimal::deserialize(deserialize!(cur.0));
-//         }
-//         Ok(maximum)
-//     }
-//
-//     fn calc_timestamp_max(
-//         ptx: &mut PrefixTransaction,
-//         aggregators_db: Database,
-//     ) -> Result<DateTime<FixedOffset>, PipelineError> {
-//         let ptx_cur = ptx.open_cursor(aggregators_db)?;
-//         let mut maximum = DateTime::<FixedOffset>::MIN_UTC;
-//
-//         // get first to get the minimum
-//         if ptx_cur.last()? {
-//             let cur = try_unwrap!(ptx_cur.read()).unwrap();
-//             maximum = match Utc.timestamp_millis_opt(i64::from_be_bytes(deserialize!(cur.0))) {
-//                 LocalResult::Single(v) => Ok(v),
-//                 LocalResult::Ambiguous(_, _) => Err(PipelineError::InternalTypeError(
-//                     TypeError::AmbiguousTimestamp,
-//                 )),
-//                 LocalResult::None => Err(PipelineError::InternalTypeError(
-//                     TypeError::InvalidTimestamp,
-//                 )),
-//             }?;
-//         }
-//         Ok(DateTime::from(maximum))
-//     }
-//
-//     fn calc_date_max(
-//         ptx: &mut PrefixTransaction,
-//         aggregators_db: Database,
-//     ) -> Result<NaiveDate, PipelineError> {
-//         let ptx_cur = ptx.open_cursor(aggregators_db)?;
-//         let mut maximum = NaiveDate::MIN;
-//
-//         // get first to get the minimum
-//         if ptx_cur.last()? {
-//             let cur = try_unwrap!(ptx_cur.read()).unwrap();
-//             maximum = NaiveDate::parse_from_str(
-//                 String::from_utf8(deserialize!(cur.0)).unwrap().as_ref(),
-//                 DATE_FORMAT,
-//             )
-//             .unwrap();
-//         }
-//         Ok(maximum)
-//     }
-//
-//     fn calc_i64_max(
-//         ptx: &mut PrefixTransaction,
-//         aggregators_db: Database,
-//     ) -> Result<i64, PipelineError> {
-//         let ptx_cur = ptx.open_cursor(aggregators_db)?;
-//         let mut maximum = i64::MIN;
-//
-//         // get first to get the maximum
-//         if ptx_cur.last()? {
-//             let cur = try_unwrap!(ptx_cur.read()).unwrap();
-//             maximum = i64::from_be_bytes(deserialize!(cur.0));
-//         }
-//         Ok(maximum)
-//     }
-//
-//     fn calc_u64_max(
-//         ptx: &mut PrefixTransaction,
-//         aggregators_db: Database,
-//     ) -> Result<u64, PipelineError> {
-//         let ptx_cur = ptx.open_cursor(aggregators_db)?;
-//         let mut maximum = u64::MIN;
-//
-//         // get first to get the maximum
-//         if ptx_cur.last()? {
-//             let cur = try_unwrap!(ptx_cur.read()).unwrap();
-//             maximum = u64::from_be_bytes(deserialize!(cur.0));
-//         }
-//         Ok(maximum)
-//     }
-// }
+fn get_max(field_hash: &HashMap<Field, u64>, return_type: FieldType) -> Result<Field, PipelineError> {
```

Comment:
This is very expensive as you sort all the keys every time. If you use a `TreeMap` you just navigate to the first key for MIN and last key for MAX

Reply:
got it!